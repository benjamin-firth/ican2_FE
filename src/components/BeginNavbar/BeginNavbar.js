import React from 'react';
import './BeginNavbar.scss';
import NavButton from '../NavButton/NavButton';
import largeLogo from '../../images/large-logo.png';

const BeginNavbar = () => {
  const buttons = [{nav: '/signup', name: 'sign up'}, {nav: '/login', name: 'log in'}];

  const renderButtons = () => {
    return buttons.map(button => (
      <NavButton nav={button.nav} name={button.name} />
    ))
  }

  return (
    <nav className='begin-nav'>
      <img className='large-logo' src={largeLogo} />
      <div className='nav-button-container'>
        {renderButtons()}
      </div>
    </nav>
  );
}

export default BeginNavbar;
