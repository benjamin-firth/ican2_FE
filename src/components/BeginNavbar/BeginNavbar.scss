@import '../../variables.scss';

.begin-nav {
  justify-content: flex-end;
  align-items: flex-end;

  button {
    margin-bottom: 2vw;
  }
}

.large-logo {
  width: 30vw;
  margin-right: 4vw;
}
