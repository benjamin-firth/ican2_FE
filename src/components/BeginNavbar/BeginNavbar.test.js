import React from 'react';
import { shallow } from 'enzyme';
import BeginNavbar from './BeginNavbar';

describe('BeginNavbar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <BeginNavbar />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
