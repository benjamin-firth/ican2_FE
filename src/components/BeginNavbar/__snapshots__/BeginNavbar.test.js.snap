// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeginNavbar should match the snapshot 1`] = `
<nav
  className="begin-nav"
>
  <img
    className="large-logo"
    src="large-logo.png"
  />
  <div
    className="nav-button-container"
  >
    <NavButton
      name="sign up"
      nav="/signup"
    />
    <NavButton
      name="log in"
      nav="/login"
    />
  </div>
</nav>
`;
